### PR TITLE
Move callbacks example to docs

### PR DESCRIPTION
**Pull request by lostella**
This way the example is run as part of the documentation build.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/mx/__init__.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 # !!! DO NOT MODIFY !!! (pkgutil-style namespace package)
 
+# flake8: noqa
```
Comment (lostella) [inline src/gluonts/mx/__init__.py:16]: @jaheba hope you don't mind this? I would avoid over-engineering these init modules if possible

Comment (jaheba) [reply src/gluonts/mx/__init__.py:16]: So what we want is to re-expose the classes and functions without sphinx noticing?

Comment (lostella) [reply src/gluonts/mx/__init__.py:16]: I'm not sure what is needed for sphinx here, I just put this annotation so that we don't need to have `# noqa` at every line (which btw complicates things when multi-line imports are there, which are sometimes enforced by `black`). I just want to avoid all this style checking to not be in the way :-)

File: docs/tutorials/mxnet_models/trainer_callbacks.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+```
+
+```python
+print(history.loss_history) # The training loss history of all 3+3 epochs we trained the model for
+```
+
+## Default callbacks
+
+In addition to the callbacks you specify, the `Trainer` class uses the two default callbacks `ModelAveraging` and `LearningRateReduction`.
+You can turn them off by setting add_default_callbacks=False when initializing the Trainer.
```
Comment (borchero) [inline docs/tutorials/mxnet_models/trainer_callbacks.md:65]: Missing code block for `add_default_callbacks=False`

File: docs/tutorials/mxnet_models/trainer_callbacks.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+```
+
+## Custom callbacks
+
+To implement your own callback you can write a class which inherits from `gluonts.mx.trainer.Callback`, and overwrite one or more of the hooks.
+Have a look at the abstract `Callback` class, the hooks take different arguments which you can use. 
+Hook methods with boolean return value stop the training if False is returned.
+
+Here is an example implementation of a Metric value based early stopping custom callback implementation.
+It only implements the hook method "on_epoch_end()" which gets called after all batches of one epoch have been processed.
```
Comment (borchero) [inline docs/tutorials/mxnet_models/trainer_callbacks.md:80]: Code block instead of quotes?

File: docs/tutorials/mxnet_models/trainer_callbacks.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+trainer=Trainer(epochs=20, add_default_callbacks=False) # use no callback at all
+```
+
+## Custom callbacks
+
+To implement your own callback you can write a class which inherits from `gluonts.mx.trainer.Callback`, and overwrite one or more of the hooks.
+Have a look at the abstract `Callback` class, the hooks take different arguments which you can use. 
+Hook methods with boolean return value stop the training if False is returned.
+
+Here is an example implementation of a Metric value based early stopping custom callback implementation.
```
Comment (borchero) [inline docs/tutorials/mxnet_models/trainer_callbacks.md:79]: "Here is an example for a custom callback implementation which terminates training early based on the value of some Metric (such as the RMSE)"?

File: src/gluonts/mx/__init__.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/mx/__init__.py:16]: The other way to suppress the warning would be to add these things to `__all__`. However, that would have the effect that sphinx autogen would include these entities in the modules documentation -- which is probably not what we want.

So I think this is the way to go.

**Review by rshyamsundar — APPROVED** [review]
